### PR TITLE
SDFSOF-64-PR3: RPC Actions: implement logic and validations

### PR DESCRIPTION
**Pull request by MazurakIhor**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

RPC actions and unit tests:

- notify_refund_sent
- notify_amounts_updated
- notify_offchain_funds_available
- notify_transaction_error
- notify_transaction_expired
- notify_transaction_recovery
- do_stellar_refund

### Why

RPC actions should replace PATCH /transactions endpoint

### CODE REVIEW

**Review by Ifropc — APPROVED** [review]

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyTransactionRecoveryHandler.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      JdbcSepTransaction txn, NotifyTransactionRecoveryRequest request) {
+    return PENDING_ANCHOR;
+  }
+
+  @Override
+  protected Set<SepTransactionStatus> getSupportedStatuses(JdbcSepTransaction txn) {
+    if (SEP_24 == Sep.from(txn.getProtocol())) {
+      JdbcSep24Transaction txn24 = (JdbcSep24Transaction) txn;
+      if (txn24.getTransferReceivedAt() != null) {
+        return Set.of(ERROR, EXPIRED);
```
[the remark this answers is not in the document]
Comment (JakeUrban) [reply platform/src/main/java/org/stellar/anchor/platform/action/NotifyTransactionRecoveryHandler.java:55]: I think I agree, as long as transactions can't expire after the business has received funds.

File: platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java [hunk 2]
```diff
@@ -117,6 +142,16 @@ private CreateTransactionRequest getCreateTransactionRequest(JdbcCustodyTransact
         .build();
   }
 
+  private CreateTransactionRequest getCreateTransactionRefundRequest(JdbcCustodyTransaction txn)
+      throws InvalidConfigException {
+    return CreateTransactionRequest.builder()
+        .assetId(fireblocksConfig.getFireblocksAssetCode(txn.getAmountAsset()))
+        .amount(txn.getAmount())
+        // TODO: set source and destination
+        // TODO: what to do with amountFee?
```
[the remark this answers is not in the document]
Comment (JakeUrban) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:151]: I think I'm missing context here. Is the fee not set on the transaction?

[the remark this answers is not in the document]
Comment (Ifropc) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:151]: It's for a refund, and I believe the fee is just for a custody transaction, not for the Platfrom transaction

File: platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java [hunk 3]
```diff
@@ -102,6 +102,31 @@ public CreateTransactionPaymentResponse createTransactionPayment(
...
+      backoff = @Backoff(delayExpression = "${custody.fireblocks.retry_config.delay}"),
+      exceptionExpression =
+          "#root instanceof T(org.stellar.anchor.api.exception.FireblocksException) AND"
+              + "(#root.statusCode == 429 OR #root.statusCode == 503)")
+  public CreateTransactionPaymentResponse createTransactionRefund(JdbcCustodyTransaction txn)
+      throws FireblocksException, InvalidConfigException {
+
+    CreateTransactionRequest request = getCreateTransactionRefundRequest(txn);
+
+    //    TODO: uncomment when getCreateTransactionRefundRequest() generates valid request
```
[the remark this answers is not in the document]
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:120]: Refund logic will most likely require additional changes in webhook handling and reconciliation job.
Full end-to-end refund logic will be implemented in scope of https://stellarorg.atlassian.net/browse/SDFSOF-65.

File: platform/src/main/java/org/stellar/anchor/platform/action/NotifyTransactionRecoveryHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/action/NotifyTransactionRecoveryHandler.java:55]: Added additional validation: status of the transaction can't be changed to EXPIRED if business has received funds.

File: platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:151]: After discussion with temmates, I understood how to use amountFee from the request
After successful processing of custody refund transaction, custody server has to send notify_refund_sent RPC action to platform with id, amount and fee. Id will be taken from stellar transaction. Amount and fee will be taken from the request.

Full implementation of refund will be done in scope of https://stellarorg.atlassian.net/browse/SDFSOF-65

File: platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ifropc) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:120]: Thanks for explanation!

File: platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ifropc) [reply platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksPaymentService.java:151]: Got it, than you!

**Pull request merged**